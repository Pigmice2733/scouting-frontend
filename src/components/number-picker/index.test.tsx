import { h, render } from 'preact'
import testUtils from 'test-utils'
import NumberPicker from '.'

test('renders', () => {
  const cb = jest.fn()
  const tree = testUtils<HTMLDivElement>(
    <NumberPicker id="asdf" onChange={cb} />
  )
  expect(tree).toMatchSnapshot()
  const [decrementButton, incrementButton] = Array.from(
    tree.querySelectorAll('button')
  )
  const input = tree.querySelector('input')
  expect(cb).not.toHaveBeenCalled()
  incrementButton.click()
  expect(cb).toHaveBeenLastCalledWith(1)
  decrementButton.click()
  decrementButton.click()
  expect(cb).toHaveBeenLastCalledWith(-1)
  input.value = '500'
  input.dispatchEvent(new Event('input'))
  expect(cb).toHaveBeenLastCalledWith(500)
  expect(cb).toHaveBeenCalledTimes(4)
})
