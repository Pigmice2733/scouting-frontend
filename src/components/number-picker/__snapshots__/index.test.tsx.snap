// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  class="numberPicker"
>
  <button>
    -
  </button>
  <input
    class="input undefined"
    type="number"
  />
  <button>
    +
  </button>
</div>
`;
